test/path.jl

```diff
@@ -339,3 +339,19 @@ end
     end
     @test isabspath(withenv(homedir, var => nothing))
 end
+
+@testset "username" begin
+    # Read from env variables
+    envvars = ("LOGNAME", "USER", "LNAME",  "USERNAME")
+    for varname in envvars
+        local uname = "testuser"
+        unset_envdict = Dict(k => "" for k in envvars if k != varname)
+        @test withenv(varname => uname, unset_envdict...) do
+            username()
+        end == uname
+    end
+
+    # Read using Libc
+    @test !isempty(username())
```

Comment:
Is it guaranteed here that `username()` does not get the value from any environment variable?

We probably also have to take special care of Windows machines (see the [test for `Filesystem.getusername`](https://github.com/JuliaLang/julia/blob/master/test/file.jl#L1684)).